reflex/ivars/base.py

```diff
@@ -1043,6 +1043,27 @@ def unionize(*args: Type) -> Type:
     return Union[first, unionize(*rest)]
 
 
+def original_var_type_or_target_type(original_var: Var, target_type: type) -> type:
+    """Inherit the original var type if possible, i.e. the original var is a subclass of the target type.
+        When the original var is a LiteralVar, the target type is always target_type.
+        When the original var is a subclass of the target type, the original var type is returned.
+
+    Args:
+        original_var: The original var of the operation.
+        target_type: The target type of the operation.
+
+    Returns:
+        The inherited type.
+    """
+    if isinstance(original_var, LiteralVar):
+        return str
```

Comment:
```suggestion
        return target_type
```

Reply:
You were right, that was a CNP one. Thanks!